hypervisor/dm/vpci/pci_pt.c

```diff
@@ -448,45 +489,65 @@ void init_vdev_pt(struct pci_vdev *vdev)
 
 	ASSERT(vdev->nr_bars > 0U, "vdev->nr_bars should be greater than 0!");
 
-	if (is_prelaunched_vm(vdev->vpci->vm)) {
-		for (idx = 0U; idx < vdev->nr_bars; idx++) {
-			pbar = &vdev->pdev->bar[idx];
-			vbar = &vdev->bar[idx];
-
-			if (is_bar_supported(pbar)) {
-				vbar->reg.value = pbar->reg.value;
-				vbar->reg.bits.mem.base = 0x0U; /* clear vbar base */
-				if (vbar->reg.bits.mem.type == 0x2U) {
-					/* Clear vbar 64-bit flag and set it to 32-bit */
-					vbar->reg.bits.mem.type = 0x0U;
-				}
+	for (idx = 0U; idx < vdev->nr_bars; idx++) {
+		pbar = &vdev->pdev->bar[idx];
+		vbar = &vdev->bar[idx];
+
+		vbar->size = 0UL;
+		vbar->reg.value = pbar->reg.value;
+		vbar->is_64bit_high = pbar->is_64bit_high;
+
+		if (pbar->is_64bit_high) {
+			ASSERT(idx > 0U, "idx for upper 32-bit of the 64-bit bar should be greater than 0!");
+
+			vbar_base = 0UL;
+			if (is_sos_vm(vdev->vpci->vm)) {
+				/* For SOS: vbar base (GPA) = pbar base (HPA) */
+				vbar_base = get_pbar_base(vdev->pdev, idx);
+			} else if (idx > 0U) {
+				/* For pre-launched VMs: vbar base is predefined in vm_config */
+				vbar_base = vdev->ptdev_config->vbar_base[idx - 1U];
+			}
```

Comment:
please add 
"else {
     // comments
}"
to this line.The else is needed in "if {} else if {} " pattern according MISRA-C standards.

Reply:
Fixed this issue and force pushed